apps/job-info-consumer/consumer/pkg/consumer.go

```diff
@@ -0,0 +1,79 @@
+package pkg
+
+import (
+	"context"
+
+	"github.com/bacalhau-project/bacalhau/pkg/node"
+	"github.com/bacalhau-project/bacalhau/pkg/pubsub"
+	"github.com/bacalhau-project/bacalhau/pkg/pubsub/libp2p"
+	"github.com/bacalhau-project/bacalhau/pkg/requester/pubsub/jobinfo"
+	"github.com/bacalhau-project/bacalhau/pkg/util"
+	libp2p_pubsub "github.com/libp2p/go-libp2p-pubsub"
+	"github.com/libp2p/go-libp2p/core/host"
+	"github.com/rs/zerolog/log"
+)
+
+type ConsumerParams struct {
+	Libp2pHost host.Host
+	Datastore  *PostgresDatastore
+}
+
+// Consumer registers to the gossipsub topic and inserts the received job info into the datastore.
+type Consumer struct {
+	libp2pHost  host.Host
+	pubSub      pubsub.PubSub[jobinfo.Envelope]
+	cleanupFunc func(context.Context)
+	datastore   *PostgresDatastore
+}
+
+func NewConsumer(params ConsumerParams) *Consumer {
+	consumer := &Consumer{
+		libp2pHost: params.Libp2pHost,
+		datastore:  params.Datastore,
+	}
+	return consumer
+}
+
+func (c *Consumer) Start(ctx context.Context) error {
+	var err error
+	ctx, cancel := context.WithCancel(ctx)
+	defer func() {
+		if err != nil {
+			cancel()
+		}
+	}()
```

Comment:
Not sure this is right since `err` is being reassigned below, we'll probably want a named return instead.